src/renderer/components/dock/terminal/terminal.ts

```diff
@@ -63,15 +59,25 @@ export class Terminal {
     const { elem } = this;
 
     if (elem) {
-      Terminal.spawningPool.appendChild(elem);
+      this.dependencies.spawningPool.appendChild(elem);
     }
   }
 
-  constructor(public tabId: TabId, protected api: TerminalApi) {
+  constructor(protected readonly dependencies: TerminalDependencies, { tabId, api }: TerminalArguments) {
+    this.tabId = tabId;
+    this.api = api;
+    const { fontSize, fontFamily } = this.dependencies.terminalConfig.get();
+
+    this.xterm = new XTerm({
```

Comment:
Side-effect?

Reply:
I don't think so... from the documentation it doesn't seem like the constructor actually mounts to anything by itself.

Reply:
This would be more apparent if there were unit tests. I think those will appear organically pretty soon, let's see about this after that. Resolved.